### PR TITLE
fix reset parameter; re-define CVBooster

### PR DESCRIPTION
**Pull request by wxchan**
@guolinke I think CVBooster is also not needed in lightgbm, because we already store dataset in Booster.

### CODE REVIEW

**Comment by guolinke** [comment]
@wxchan OK, I think you can remove it as well.

**Comment by wxchan** [comment]
@guolinke I re-define CVBooster from one booster of one fold to all boosters of all folds, I can call all the methods of Booster with CVBooster, so I do not need to distinguish model and cvfolds.

File: python-package/lightgbm/callback.py [hunk 1]
```diff
@@ -121,13 +121,20 @@ def reset_parameter(**kwargs):
...
             if isinstance(value, list):
                 if len(value) != env.end_iteration - env.begin_iteration:
                     raise ValueError("Length of list {} has to equal to 'num_boost_round'.".format(repr(key)))
-                env.model.reset_parameter({key: value[env.iteration - env.begin_iteration]})
+                new_param = value[env.iteration - env.begin_iteration]
             else:
-                env.model.reset_parameter({key: value(env.iteration - env.begin_iteration)})
+                new_param = value(env.iteration - env.begin_iteration)
+            if new_param != env.params.get(key, None):
+                new_parameters[key] = new_param
```
Comment (guolinke) [inline python-package/lightgbm/callback.py]: I think you should update env.params after reset_parameters.

Comment (wxchan) [inline python-package/lightgbm/callback.py]: Fixed.

**Pull request merged**